debian/cloud-init.preinst

```diff
@@ -1,237 +1,39 @@
-        # from 14.04 to 16.04
-        return 0
-
-     elif [ -z "${new_instance_id}" -o -z "${old_instance_id}" ]; then
-        cat <<EOM
-
-WARNING: Failed to migrate old instance ID to new instance ID.
-    Cloud-init may detect this instance as a new instance upon reboot.
-    Please see: https://bugs.launchpad.net/bug/1506187
-
-EOM
-
-     elif [ "${new_instance_id}" != "${old_instance_id}" ]; then
-        cat <<EOM
-
-AZURE: this instance uses an unstable instance ID. Cloud-init will
-    migrate the instance ID from:
-            ${old_instance_id}
-        to:
-            ${new_instance_id}
-    For more information about this change, please see:
-    https://bugs.launchpad.net/bug/1506187
-    https://azure.microsoft.com/en-us/blog/accessing-and-using-azure-vm-unique-id
-
-EOM
-
-        # Write the new instance id
-        echo "${new_instance_id}" > /var/lib/cloud/data/instance-id
-
-        # Remove the symlink for the instance
-        rm /var/lib/cloud/instance
-
-        # Rename the old instance id to the new one
-        mv /var/lib/cloud/instances/${old_instance_id} \
-           /var/lib/cloud/instances/${new_instance_id}
-
-        # Link the old id to the new one, just incase
-        ln -s /var/lib/cloud/instances/${new_instance_id} \
-              /var/lib/cloud/instances/${old_instance_id}
-
-        # Make the active instance the new id
-        ln -s /var/lib/cloud/instances/${new_instance_id} \
-              /var/lib/cloud/instance
-  fi
-fi
-}
-
-cleanup_lp1552999() {
-    local oldver="$1" last_bad_ver="0.7.7~bzr1178"
-    dpkg --compare-versions "$oldver" le "$last_bad_ver" || return 0
-    local hdir="/var/lib/systemd/deb-systemd-helper-enabled"
-    hdir="$hdir/multi-user.target.wants" 
-    local edir="/etc/systemd/system/multi-user.target.wants"
-    rm -f "$hdir/cloud-config.service" "$hdir/cloud-final.service" \
-        "$hdir/cloud-init-local.service" "$hdir/cloud-init.service"
-}
-
 cleanup_oci_network_lp1956788() {
    # Remove vestigial /etc/cloud/cloud.cfg.d/99-disable-network-config.cfg
-   # from Oracle now that datasource honors system cfg above datasource cfg.
+   local oldver="$1" last_bad_ver="24.3~1g6e4153b3-0ubuntu1"
+   dpkg --compare-versions "$oldver" le "$last_bad_ver" || return 0  # from Oracle now that datasource honors system cfg above datasource cfg.
    grep DataSourceOracle /var/lib/cloud/instance/datasource > /dev/null 2>&1 || return 0
    rm -f /etc/cloud/cloud.cfg.d/99-disable-network-config.cfg
 }
 
+cleanup_user_scripts() {
+   local oldver="$1" last_bad_ver="24.3~1g6e4153b3-0ubuntu1"
+   dpkg --compare-versions "$oldver" le "$last_bad_ver" || return 0  # from Oracle now that datasource honors system cfg above datasource cfg.
+   # 0.6.0 changed 'user-scripts' to config-scripts-user (LP: #1049146)
+   if [ -e /var/lib/cloud/instance/sem/user-scripts ]; then
+      ln -sf user-scripts /var/lib/cloud/instance/sem/config-scripts-user
+   fi
+}
+
+cleanup_rename_config() {
```

Comment:
Any reason we can't also kill this? I think even back to Trusty we have `90_dpkg.cfg` and no `distro.cfg`.

Reply:
Agreed